### PR TITLE
Bitmap marking

### PR DESCRIPTION
**Pull request by nyuichi**
Experimental implementation of bitmap marking. Please DO NOT merge this pull request since performance issues have not yet been resolved. The major cause is a newly introduced function called `obj2page`, which resolves pic_object to the page it belongs to. O(1) algorithm is needed for merging, although I don't have ideas right now.

### CODE REVIEW

**Comment by KeenS** [comment]
FYI, in [the original paper about Bitmap GC](http://www.pllab.riec.tohoku.ac.jp/papers/icfp2011UenoOhoriOtomoAuthorVersion.pdf), each page is allocated at power of 2 boundary address so you can get the page address by clearing lower bits of obj address.

**Comment by wasabiz** [comment]
@KeenS 

Wow! Sounds good!

File: extlib/benz/include/picrin/compat.h [hunk 1]
```diff
@@ -23,6 +23,20 @@ extern "C" {
 # define offsetof(s,m) ((size_t)&(((s *)NULL)->m))
 #endif
 
+#if __STDC_VERSION__ >= 199901L
+# include <stdint.h>
+#else
+# if INT_MAX > 2147483640L      /* borrowed from luaconf.h */
+typedef int int32_t;
+typedef unsigned int uint32_t;
+# elif LONG_MAX > 2147483640L
```
Comment (omasanori) [inline extlib/benz/include/picrin/compat.h]: ISO C requires that LONG_MAX is at least 2147483647, so this ``elif ...`` can replace with ``else`` unless we are willing to support insane platforms.

Comment (wasabiz) [inline extlib/benz/include/picrin/compat.h]: thanks!

**Pull request merged**